src/main/java/net/snowflake/client/core/HttpClientSettingsKey.java

```diff
@@ -15,35 +15,34 @@
 public class HttpClientSettingsKey implements Serializable {
 
   private OCSPMode ocspMode;
-  private boolean useProxy;
+  private boolean useProxy = false;
   private String proxyHost = "";
   private int proxyPort = 0;
   private String nonProxyHosts = "";
   private String proxyUser = "";
   private String proxyPassword = "";
   private String proxyProtocol = "http";
+  private String userAgentIdentifier = "";
 
   public HttpClientSettingsKey(
       OCSPMode mode,
+      boolean useProxy,
       String host,
       int port,
       String nonProxyHosts,
       String user,
       String password,
-      String scheme) {
-    this.useProxy = true;
+      String scheme,
+      String userPrefix) {
+    this.useProxy = useProxy;
     this.ocspMode = mode != null ? mode : OCSPMode.FAIL_OPEN;
     this.proxyHost = !Strings.isNullOrEmpty(host) ? host.trim() : "";
     this.proxyPort = port;
     this.nonProxyHosts = !Strings.isNullOrEmpty(nonProxyHosts) ? nonProxyHosts.trim() : "";
     this.proxyUser = !Strings.isNullOrEmpty(user) ? user.trim() : "";
     this.proxyPassword = !Strings.isNullOrEmpty(password) ? password.trim() : "";
     this.proxyProtocol = !Strings.isNullOrEmpty(scheme) ? scheme.trim() : "http";
-  }
-
-  public HttpClientSettingsKey(OCSPMode mode) {
```

Comment:
Since this overall class is marked public, technically it's not safe to remove this in case external users already use this interface. But I suppose it's not 100% clear whether we intend this to be an official part of the real public interface regardless of Java-level visibility.

Hadoop had run into similar problems before when too many things were unintentionally "public" by default and users took dependencies on internal implementation details.

To play it safe, it's probably better to keep this but mark it deprecated.

I defer to the JDBC folks for the final decision though -- if we're not too strict about semantic versioning and breaking changes in this library then maybe there's already precedent for removing things like this.

Reply:
Although I agree with you in principle, I see that HttpClientSettingsKey is already being used outside the net.snowflake.client.core package in the net.snowflake.cleint.jdbc package.
Regarding depreciated annotation, I will check with JDBC folks on what's their guideline.

Reply:
Keeping the publicly available constructor while creating another one to use for builder pattern.